lib/client.js

```diff
@@ -398,6 +465,15 @@ MqttClient.prototype.publish = function (topic, message, opts, callback) {
     dup: opts.dup
   }
 
+  if (this.options.protocolVersion === 5) {
+    packet.properties = opts.properties
+    if ((!this.options.properties && packet.properties && packet.properties.topicAlias) || ((opts.properties && this.options.properties) &&
+      ((opts.properties.topicAlias && this.options.properties.topicAliasMaximum && opts.properties.topicAlias > this.options.properties.topicAliasMaximum) ||
+        (!this.options.properties.topicAliasMaximum && opts.properties.topicAlias)))) {
+      delete packet.properties.topicAlias
+    }
+  }
```

Comment:
Can you please add a comment explaining this block?
Also, can you extract `var options = this.options`  for readability?

Reply:
fixed